### PR TITLE
Fix uploading big files to the media library

### LINKED ISSUE #2998: Embedding mp4 video from MediaLibrary doesn't work
## Bug Report

| Q            | A     |
| ------------ | ----- |
| Critical Bug | no    |
| Version      | tested with 5.4.0, 5.7.0 and whatever is installed on demo.fork-cms.com |

### Problem description
It's possible to upload an mp4 video the the media library, but when i go to a page and edit an "editor" block, trying to insert the video fails.
I can click the "Browse Server" button and select the file, but when i then click on OK i get an error message "please provide a valid URL"

### Steps to reproduce
(can be done on demo.fork-cms.com)
- Upload any mp4 video file to media library, such as this https://sample-videos.com/video123/mp4/720/big_buck_bunny_720p_1mb.mp4
- Go and edit any page, click "Embed Media"
- Click "Browse Server" and select the video you just uploaded from your library
- Try and save it by pressing OK

### Expected behavior / Proposed solution
I would expect the video the become embedded. The path that gets determined is correct. As a workaround i inserted a video tag myself through the source code view, and just copy+pasted the path. Video is now visible on the page

I'm pretty sure that the source of error is in lines 295-298 of src/Backend/Core/Js/ckeditor/plugins/oembed/plugin.js

As a workaround, i tried to just prepend https://fork.installation.url/ but to no avail

### LINKED ISSUE #3133: How to upload audio files?
## Bug Report

| Q            | A     |
| ------------ | ----- |
| Critical Bug | no    |
| Version      | latest |

### Problem description

At the media overview page in the backend (http://demo.forkcms.com/private/en/media_library/media_item_index) there are tabs for the different kinds of media you have (files, videos, images, audio). Files, video and images can be uploaded if you click the 'add media' button. However, I can't seem to find a way to **upload audio files**. 

### Expected behavior / Proposed solution

Either make it possible to upload audio files, or get rid of the 'audio' tab.

### PR DESCRIPTION
**Pull request by carakas**
## Type

- Non critical bugfix

## Resolves the following issues

fixes #3133 
fixes #2998 

## Pull request description

Turns out the chunking of large files didn't work in the media library.
This pr fixes that issue so you can upload video and audio files

Screenshot 2020-07-04 at 11 28 52

### CODE REVIEW

File: src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php [hunk 1]
```diff
@@ -5,25 +5,73 @@
...
+        $this->fileManager = $this->get('media_library.manager.file');
+        $this->extensionManager = $this->get('media_library.manager.extension');
+        $this->mimeTypeManager = $this->get('media_library.manager.mime_type');
+        $this->mediaFolderRepository = $this->get('media_library.repository.folder');
+        $this->commandBus = $this->get('command_bus');
+    }
+
     /**
      * PHP Server-Side Example for Fine Uploader (traditional endpoint handler).
      * Maintained by Widen Enterprises.
```
Comment (jessedobbelaere) [inline src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php:77]: We should probably remove this messagesince a lot of code changed :P

File: src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php [hunk 2]
```diff
@@ -56,87 +104,45 @@ public function execute(): void
...
 
-        // Generate folder if not exists
-        $this->get('media_library.manager.file')->createFolder($uploadDir);
-
-        // Call handleUpload() with the name of the folder, relative to PHP's getcwd()
-        $result = $uploader->handleUpload($uploadDir);
+        $result = $this->uploadHandler->handleUpload($this->uploadDirectory);
 
-        if (array_key_exists('error', $result)) {
+        if (array_key_exists('error', $result)
```
Comment (jessedobbelaere) [inline src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php:115]: Personal preference: would extract if-logic to separate method that returns bool, for readability

Comment (carakas) [reply src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php]: 9303ba4e592f14461862c2aeda4aacd3dce78595

File: src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php (hunk 1 above)
Comment (carakas) [reply src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php]: removed in 3ef5b6992690eb67f0cfba70b71539223f9c8d59

File: src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php (hunk 2 above)
Comment (jessedobbelaere) [reply src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php]: awesome

**Pull request merged**